src/main/kotlin/com/hibob/academy/employeeFeedback/service/FeedbackService.kt

```
@@ -0,0 +1,70 @@
+        val minCharacters = 30
+        val maxCharacters = 500
+        return !(feedback.length < minCharacters || feedback.length > maxCharacters)
+    }
+
+    fun anonymityStatusValidation(employeeId: Int?): Int{
+        return employeeId ?: throw Exception("Can't check status of anonymous feedback")
+    }
+
+    fun authorizationToViewValidation(employeeId: Int, feedbacksEmployeeId: Int) {
+        if (feedbacksEmployeeId != employeeId)
+            throw BadRequestException("Unauthorized to view this feedback status")
+    }
+
+    fun submitFeedback(employeeId: Int, companyId: Int, anonymity: AnonymityType, feedback: String): Boolean {
+        var employeeIdForFeedback: Int? = employeeId
+
+        if (anonymity == AnonymityType.ANONYMOUS)
+            employeeIdForFeedback = null
+
+        val validation = feedbackLengthValidation(feedback)
+
+        if (validation) {
+            val feedbackForSubmission = FeedbackForSubmission(
+                employeeIdForFeedback,
+                companyId,
+                anonymity,
+                feedback)
+            feedbackDao.submitFeedback(feedbackForSubmission)
+
+            return true
+        }
+
+        return false
+    }
+
+    fun viewAllSubmittedFeedback(employeeId: Int, companyId: Int): List<FeedbackInfo> {
+        return feedbackDao.viewAllSubmittedFeedback(companyId)
+    }
+
+    fun viewStatusOfMyFeedback(employeeId: Int, companyId: Int, feedbackId: Int): Map<Int, Boolean> {
+        val feedbackInfo = feedbackDao.getFeedbackById(feedbackId, companyId)
+
+        feedbackInfo?.let{
+
+            val feedbacksEmployeeId = anonymityStatusValidation(feedbackInfo.employeeId)
+            authorizationToViewValidation(employeeId, feedbacksEmployeeId)
+
+            val feedbackStatus = FeedbackStatusData(companyId, employeeId, feedbackId)
+
+            return feedbackDao.viewStatusOfMyFeedback(feedbackStatus)
+
+        } ?: throw Exception("Feedback not found")
```

better to throw an notFound exception instead of a generic one